src/falcon/mod.rs

```diff
@@ -0,0 +1,24 @@
+mod ffi;
+mod hash;
+mod keys;
+mod polynomial;
+mod signature;
+
+pub use keys::{KeyPair, PublicKey, SecretKey};
+pub use polynomial::Polynomial;
+pub use signature::Signature;
+
+// CONSTANTS
+// ================================================================================================
+
+const MODULUS: u16 = 12289;
+const MODULUS_MINUS_1_OVER_TWO: u16 = 6144;
+
+const LOG_N: usize = 9;
+const N: usize = 512;
+
+const PK_LEN: usize = 897;
+const SK_LEN: usize = 1281;
+const SIG_LEN: usize = 666;
+const SIG_L2_BOUND: u64 = 34034726;
+const KEYGEN_BUF_LEN: usize = 15879;
```

Comment:
nit: I wonder if it would make sense to define type aliases based on these rather than passing constants around. For example:
```Rust
type PublicKeyBytes = [u8; PK_LEN];
type SecretKeyBytes = [u8; SK_LEN];
type SignatureBytes = [u8; SIG_LEN];
```

Reply:
Makes sense! We might still need the constants though but I still think that it is worth it to use type aliases in order to improve code readability.